### PR TITLE
ci: add editorconfig and prettier

### PR DESCRIPTION
**Pull request by balupton**
Non-functional changes have been applied automatically via editorconfig and prettier, enforcing consistent indentation, line endings, and in some cases code cleanups.

Non-functional changes are:

- Added package.json to ensure consistent prettier version is used between us
- Added installation of prettier via `npm install` to `make setup`
- Added prettier to `make {fmt,lint}`

This knocks off these meta issues for this repository:

- https://github.com/textileio/meta/issues/4
- https://github.com/textileio/meta/issues/3

### CODE REVIEW

File: qa/wallet.sh [hunk 1]
```diff
@@ -115,132 +116,130 @@ tpc0=$(echo ${test0} | jq ".peer_count")
...
+	export API=$(getApi i)
+	textile blocks ls -t $(textile config Account.Thread) -l 100 -d | dot -Tpng -o /tmp/peer${i}_account_blocks.png
+	textile blocks ls -t ${thread} -l 100 -d | dot -Tpng -o /tmp/peer${i}_thread_blocks.png
 done
 
 echo "Success."
 
-if [[ "$wait" = "yes" ]]; then
-    wait
+if [[ "$wait" == "yes" ]]; then
```
Comment (balupton) [inline qa/wallet.sh]: Seems like it caught a bug here.

Comment (sanderpick) [reply qa/wallet.sh]: https://stackoverflow.com/a/2237103/641834

Comment (sanderpick) [reply qa/wallet.sh]: New conventions? It was working fine... maybe not in newer shells.

Comment (balupton) [reply qa/wallet.sh]: I think we may need to adopt https://github.com/textileio/meta/issues/2 as the different shell scripts in this repo are using different shells.

This particular file is using bash, despite the `.sh` extension.

Bash treats `=` and `==` the same within square brackets, so the formatter swapped it for the option that reduces user error I guess.

So this is a non-functional change as expected.

We should take steps though to ensure our shell scripts are consistent, ref:

- https://github.com/textileio/meta/issues/5
- https://github.com/textileio/meta/issues/2
- https://github.com/textileio/meta/issues/1

Comment (sanderpick) [reply qa/wallet.sh]: Totally. Wow, I'm sold on the formatter.

**Comment by balupton** [comment]
@sanderpick once merged, you should add this plugin to your code editor:

- https://plugins.jetbrains.com/plugin/10456-prettier

@carsonfarmer @asutula @andrewxhill for us VSCode users, make sure you have this plugin installed:

- https://marketplace.visualstudio.com/items?itemName=EditorConfig.EditorConfig

**Pull request merged**